crates/gitbutler-core/src/virtual_branches/virtual.rs

```diff
@@ -833,19 +838,18 @@ pub fn list_virtual_branches(
             })
         })?;
 
-    let (statuses, skipped_files) = get_status_by_branch(gb_repository, project_repository)?;
+    let integration_commit =
+        super::integration::update_gitbutler_integration(gb_repository, project_repository)?;
+
+    let (statuses, skipped_files) =
+        get_status_by_branch(gb_repository, project_repository, Some(&integration_commit))?;
     let max_selected_for_changes = statuses
         .iter()
         .filter_map(|(branch, _)| branch.selected_for_changes)
         .max()
         .unwrap_or(-1);
 
     for (branch, files) in &statuses {
-        // check if head tree does not match target tree
-        // if so, we diff the head tree and the new write_tree output to see what is new and filter the hunks to just those
-        let files =
-            calculate_non_commited_diffs(project_repository, branch, &default_target, files)?;
```

Comment:
Trying to test this out and understand, does skipping this work? Or does `get_status_by_branch` give us something else now?

Reply:
By producing its own input, I mean it did something complicated that returned its original input. That and it resolved conflict markers, I missed that, but also just fixed it.